### DISCUSSION TITLE
How connect javascript auth object with desktop firebase auth object?

### DISCUSSION THREAD

**Discussion by filipenanclarez** [invertase/flutterfire_desktop#73] [Q&A]
Hi, 

Since [we don't have a way to make auth with apple works in windows](https://github.com/invertase/flutter_desktop_webview_auth/issues/18), then i make this in javascript page to use inside [windows webview](https://pub.dev/packages/webview_windows).

Now a want to send my auth object to firebase object on the dart side.

I already can use channels to comunicate between webview and flutter app.

But i dont know how send auth data to existing firebase auth object to conect them.

How can'i do this?

**Comment by pr-Mais** [comment]
I believe it's going to be the same as [the official example here](https://firebase.google.com/docs/auth/flutter/federated-auth#apple), you can use `OAuthProvider`, create a credential object then pass it to `signInWithCredential`.
